File: bank-service/src/main/java/rs/edu/raf/dto/CurrencyCreateDTO.java
package rs.edu.raf.dto;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;

public record CurrencyCreateDTO(@NotNull @NotEmpty String name, @NotNull @NotEmpty String code,
                                @NotNull @NotEmpty String symbol, @NotNull Long countryId) {
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/dto/CurrencyCreateDTO.kt
package org.example.bankservicekotlin.dto

import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.NotNull

data class CurrencyCreateDTO(
    @NotNull @NotEmpty val name: String,
    @NotNull @NotEmpty val code: String,
    @NotNull @NotEmpty val symbol: String,
    @NotNull val countryId: Long
)
